fk/secretreceive.go

```diff
@@ -202,13 +230,77 @@ func countDigits(i int) (count int) {
 	return len(iString)
 }
 
+func getFileNameToWrite(filename string) (string, error) {
+	userDir, err := homedir.Dir()
+	if err != nil {
+		return "", err
+	}
+	downloadsDir := userDir + "/Downloads"
+	fileInfo, err := os.Stat(downloadsDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return "", fmt.Errorf(downloadsDir + " directory doesn't exist")
+		} else {
+			return "", err
+		}
+	}
+	if !fileInfo.IsDir() {
+		return "", fmt.Errorf(fileInfo.Name() + " file exists and is not a directory")
+	}
+	filenameToWrite := getNewUniqueFilename(
+		filepath.FromSlash(downloadsDir+"/"+filename),
+		doesntExist,
+	)
+
+	return filenameToWrite, nil
+}
+
+func getNewUniqueFilename(fp string, checker func(string) bool) string {
+	i := 0
+	foundFilename := false
+	filename := filepath.Base(fp)
+	numberedFilename := filename
+	for ok := true; ok; ok = (foundFilename == false) {
+		if i == 0 {
```

Comment:
Rather than having the `0` case exception *inside* the for loop you might want to just have a check before it (and before you set up `i`, `numberedFilename` e.g.:

```
if !fileExists(filenamePassedIntoFunction) {
    return filenamePassedIntoFunction
}
```